metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -1189,5 +1291,56 @@ def get_data_platform_instance() -> DataPlatformInstanceClass:
     def get_report(self):
         return self.report
 
+    def create_checkpoint(self, job_id: JobId) -> Optional[Checkpoint]:
+        """
+        Create the custom checkpoint with empty state for the job.
+        """
+        assert self.ctx.pipeline_name is not None
+        if job_id == self.get_default_ingestion_job_id():
+            return Checkpoint(
+                job_name=job_id,
+                pipeline_name=self.ctx.pipeline_name,
+                platform_instance_id=self.get_platform_instance_id(),
+                run_id=self.ctx.run_id,
+                config=self.source_config,
+                # Reusing BaseSQLAlchemyCheckpointState as it has needed functionality to support statefulness of GLue
+                state=BaseSQLAlchemyCheckpointState(),
+            )
+        return None
+
+    def get_platform_instance_id(self) -> str:
+        assert self.source_config.platform_instance is not None
```

Comment:
same comment: why is this mandatory?

Reply:
The idea of this method is to obtain the source identifier. If there are two sources with the same platform, the property `platform_instance` is what make them identifiable. However, to make stateful ingestion more flexible I have changed the implementation of this method making platform instance optional.